asyncio_mqtt/client.py

```diff
@@ -21,13 +21,21 @@
     Type,
     Union,
     cast, Iterable,
+    AsyncContextManager,
+    TypeVar,
 )
 
 
+
 try:
     from contextlib import asynccontextmanager
 except ImportError:
-    from async_generator import asynccontextmanager  # type: ignore
+    from async_generator import asynccontextmanager as _asynccontextmanager # type: ignore
+    from typing_extensions import ParamSpec
+    _P = ParamSpec('_P')
+    _T = TypeVar('_T')
+    def asynccontextmanager(func: Callable[_P, AsyncIterator[_T]]) -> Callable[_P, AsyncContextManager[_T]]: # type: ignore
+        return _asynccontextmanager(func)
```

Comment:
https://github.com/sbtinstruments/asyncio-mqtt/pull/87 has a less-verbose fix for this for python3.7+

I suspect some combination of the two is desired for support for 3.6 with type information.

Reply:
i did test this with 3.6, though not with a 3.6 pyright (I don't think it would make a difference)

this basically just copies the exact signature that the contextlib version has, so theres no need for pyright to understand sys.version, I don't think.

maybe it would still be better to use sys.version instead of the try-except though.